gems/aws-sdk-core/lib/seahorse/client/net_http/handler.rb

```diff
@@ -79,8 +79,16 @@ def transmit(config, req, resp)
               bytes_received = 0
               resp.signal_headers(status_code, headers)
               net_resp.read_body do |chunk|
+                if bytes_received < resp.body.size
```

Comment:
Can you talk a bit about the motivation of this part of the change? I can follow what you're doing, but want to check about any possible unintended consequences.

Reply:
Imagine we're downloading an S3 object to a file. Seahorse will append each chunk to the file object as it is being downloaded. However, if a timeout error occurs, currently Seahorse will truncate the incomplete file object, retry the request, and start writing to the file from the beginning. This is not efficient, because there is nothing wrong with what we've already downloaded.

This change detects when the request has been retried and remembers how many bytes have already been written, and when it starts downloading again, instead of truncating the file and writing from the beginning, it will simply skip the data that was already written and start sending only new chunks. As a side effect, this allows streaming downloads (where block is provided as a target) to be retried in case of timeouts, where previously they were not retried, as a block is not "truncatable".

Ideally when we retry the request, we should make a `Range` request to ask S3 only for the remainder of the data that we haven't managed to download. I plan to make that change next, but I thought it's good to have this safety net for when that particular AWS endpoint doesn't support `Range` requests (since Seahorse is used throughout all the AWS services, if I understood correctly).